### ISSUE TITLE
brew audit exception on head-only formula

### ISSUE THREAD

**Issue by yurikoles** [Homebrew/brew#4345]
- [x] are reporting a bug others will be able to reproduce and not asking a question. If you're not sure or want to ask a question do so on our Discourse: https://discourse.brew.sh
- [x] ran a `brew` command and reproduced the problem with multiple formulae? If it's a problem with a single, official formula (not cask) please file this issue at Homebrew/homebrew-core: https://github.com/Homebrew/homebrew-core/issues/new/choose. If it's a `brew cask` problem please file this issue at https://github.com/Homebrew/homebrew-cask/issues/new/choose. If it's a tap (e.g. Homebrew/homebrew-php) problem please file this issue at the tap.
- [x] ran `brew update` and can still reproduce the problem?
- [x] ran `brew doctor`, fixed all issues and can still reproduce the problem?
- [x] ran `brew config` and `brew doctor` and included their output with your issue?

I had created head-only formula [`yurikoles/yurikoles/shaderc`](https://github.com/yurikoles/homebrew-yurikoles/blob/master/Formula/shaderc.rb) which doesn't specify any version. 
I think `brew audit` should produce something other then fail with exception.

```
brew audit --strict --online shaderc
Error: undefined method `version' for nil:NilClass
Please report this bug:
  https://docs.brew.sh/Troubleshooting
/usr/local/Homebrew/Library/Homebrew/dev-cmd/audit.rb:587:in `block in audit_specs'
/usr/local/Homebrew/Library/Homebrew/dev-cmd/audit.rb:586:in `map'
/usr/local/Homebrew/Library/Homebrew/dev-cmd/audit.rb:586:in `audit_specs'
/usr/local/Homebrew/Library/Homebrew/dev-cmd/audit.rb:862:in `block in audit'
/usr/local/Homebrew/Library/Homebrew/dev-cmd/audit.rb:855:in `each'
/usr/local/Homebrew/Library/Homebrew/dev-cmd/audit.rb:855:in `audit'
/usr/local/Homebrew/Library/Homebrew/dev-cmd/audit.rb:126:in `block in audit'
/usr/local/Homebrew/Library/Homebrew/dev-cmd/audit.rb:122:in `each'
/usr/local/Homebrew/Library/Homebrew/dev-cmd/audit.rb:122:in `audit'
/usr/local/Homebrew/Library/Homebrew/brew.rb:99:in `<main>'
```

**Comment by apjanke** [comment]
> I think `brew audit` should produce something other then fail with exception.

Right you are. `brew audit` should never throw exceptions in response to formula issues.

Looking in to this now.

**Comment by apjanke** [comment]
Here's a fix: #4349.

Note that you'll still get an audit complaint about it being head-only, but now it's a readable complaint, and it will still do the rest of the audit.
